src/Construction/UnserializeObjectConstructor.php

```diff
@@ -19,6 +19,10 @@ final class UnserializeObjectConstructor implements ObjectConstructorInterface
      */
     public function construct(DeserializationVisitorInterface $visitor, ClassMetadata $metadata, $data, array $type, DeserializationContext $context): ?object
     {
+        if ($metadata->isEnum) {
+            return (new \ReflectionEnum($type['name']))->getCase($data['name'])->getValue();
```

Comment:
Can you add a backed enum test case?

My code requires this change:

```suggestion
            return (new \ReflectionEnum($type['name']))->getCase($data)->getValue();
```

(for

```php
enum FlopNetworkEventType: int
{
    case UPDATE = 1;
    case DELETE = 2;
}
```
)